### PR TITLE
Upgrade to nexus api 5.1.2-01

### LINKED ISSUE #146: [FEATURE] Upgrade nexus-platform-api
* **What are you trying to do?**
Keep [nexus-platform-api](https://central.sonatype.com/artifact/com.sonatype.nexus/nexus-platform-api/) up to date.

Currently used version is 4.0.7-01

* **What feature or behavior is this required for?**
Each version of that library provides bug fixes, improvements and new features when scanning and evaluating applications using Sonatype Lifecycle (IQ Server).

There are particularities in this versions:

- 4.0.8-01: doesn't seem to include Logback as logger implementation so nothing gets printed.
- 4.1.4-01: comes with Jackson 2.15.1 as dependency. From Jackson 2.15.0-rc1 there is a problem when depending on that library in plugins if the target project is using a Gradle version lower than 7.6.2 and this plugin currently is compatible with Gradle 3.3 or higher.
  - This is not a problem in Jackson or in this plugin but a known issue in Gradle. See:
    - https://github.com/FasterXML/jackson-core/issues/955
    - https://github.com/gradle/gradle/issues/24390
    - https://discuss.gradle.org/t/is-there-any-way-to-skip-gradle-instrumenting-classpath-file-transformer/45213/3
- 4.3.0-01: introduces a new method related to [Legacy Violations](https://help.sonatype.com/iqserver/product-information/release-notes#ReleaseNotes-EmbracingInclusionwithLegacyViolations).
- 5.0.0-01: requires Java 11.

* **How could we solve this issue? (Not knowing is okay!)**
- We'd need to stop supporting Gradle versions lower than 7.6.2
- Experiment with Gradle Toolchain and see if we can compile the plugin with Java 11 and keep it working on projects using Java 8. See https://discuss.gradle.org/t/compile-with-java-11-to-java-8-language-level/39123/5
  - If that fails, then we'd need to stop supporting projects using a Java version lower than 11.

cc @bhamail / @DarthHater / @shaikhu

### PR DESCRIPTION
**Pull request by shaikhu**
* Upgrade `nexus-platform-api` dependency to ~5.0.0.-01~ 5.1.2-01
* Upgradle gradle wrapper to 7.6.4

It relates to the following issue:
* Fixes #146

### CODE REVIEW

File: src/main/java/org/sonatype/gradle/plugins/scan/nexus/iq/scan/NexusIqScanTask.java [hunk 1]
```diff
@@ -208,7 +208,7 @@ else if (Action.ID_WARN.equals(actionTypeId)) {
         applicationPolicyEvaluation.getCriticalComponentCount(), applicationPolicyEvaluation.getSevereComponentCount(),
         applicationPolicyEvaluation.getModerateComponentCount()));
     message.append(String.format("Number of legacy violations: %s\n",
-        applicationPolicyEvaluation.getGrandfatheredPolicyViolationCount()));
+        applicationPolicyEvaluation.getLegacyViolationCount()));
```
Comment (shaikhu) [inline src/main/java/org/sonatype/gradle/plugins/scan/nexus/iq/scan/NexusIqScanTask.java:211]: Use the new/renamed method for legacy violations

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.4-bin.zip
```
Comment (shaikhu) [inline gradle/wrapper/gradle-wrapper.properties:3]: I attempted to upgrade to gradle 8.8, but ran into breaking changes. If we want to upgrade to 8.8 we should tackle it in anothor PR.

Note using 7.6.4 works, but does prints the error `Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.`

File: build.gradle [hunk 3]
```diff
@@ -53,10 +53,7 @@ allprojects {
 }
 
 dependencies {
-  implementation ("com.sonatype.nexus:nexus-platform-api:$nexusPlatformApiVersion") {
-    exclude group: 'org.eclipse.jgit', module: 'org.eclipse.jgit'
-  }
-  implementation 'org.eclipse.jgit:org.eclipse.jgit:5.13.3.202401111512-r'
+  implementation "com.sonatype.nexus:nexus-platform-api:$nexusPlatformApiVesion"
```
Comment (shaikhu) [inline build.gradle:56]: No longer need to exclude `jgit` and explicitly define the version. Version pulled in through `nexus-platform` 5.0.0.-01 is safe.

File: build.gradle [hunk 4]
```diff
@@ -53,10 +53,7 @@ allprojects {
 }
 
 dependencies {
-  implementation ("com.sonatype.nexus:nexus-platform-api:$nexusPlatformApiVersion") {
-    exclude group: 'org.eclipse.jgit', module: 'org.eclipse.jgit'
-  }
-  implementation 'org.eclipse.jgit:org.eclipse.jgit:5.13.3.202401111512-r'
+  implementation "com.sonatype.nexus:nexus-platform-api:$nexusPlatformApiVersion"
```
Comment (shaikhu) [inline build.gradle:56]: No longer need to exclude `jgit` and explicitly define the version. The version pulled in through `nexus-platform` 5.0.0.-01 is safe.

File: gradle/integration-test.gradle [hunk 5]
```diff
@@ -12,11 +12,7 @@ task integrationTest(type: Test) {
   group = 'verification'
 }
 
-['**/ScanIT_Gradle_Versions_5_0*.class',
- '**/ScanIT_Gradle_Versions_5_5*.class',
- '**/ScanIT_Gradle_Versions_6_0*.class',
- '**/ScanIT_Gradle_Versions_6_6*.class',
- '**/ScanIT_Gradle_Versions_7_0*.class'].eachWithIndex { className, index ->
```
Comment (shaikhu) [inline gradle/integration-test.gradle:19]: Had to delete these tests as mentioned in this [comment](https://github.com/sonatype-nexus-community/scan-gradle-plugin/issues/146#issue-2053155602) "_We'd need to stop supporting Gradle versions lower than 7.6.2_".

File: gradle.properties [hunk 6]
```diff
@@ -18,7 +18,7 @@ group=org.sonatype.gradle.plugins
 version=3.0.0-SNAPSHOT
 release.useAutomaticVersion=true
 
-nexusPlatformApiVersion=4.0.7-01
+nexusPlatformApiVersion=5.0.0-01
```
Comment (shaikhu) [inline gradle.properties:21]: Upgrading to 5..0.0-01 prints deprecation warnings for usage of the cyclonedx class `Tool`.  Not familiar with cyclonedx so not sure what the recommended class is to use, or timeline for adoption.

See usages [here ](https://github.com/sonatype-nexus-community/scan-gradle-plugin/blob/a4409e452abf9859b5b5958019039e19a52d61dc/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/CycloneDxResponseHandler.java#L150)and [here](https://github.com/sonatype-nexus-community/scan-gradle-plugin/blob/a4409e452abf9859b5b5958019039e19a52d61dc/src/test/java/org/sonatype/gradle/plugins/scan/ossindex/CycloneDxResponseHandlerTest.java#L268).

File: .circleci/config.yml [hunk 7]
```diff
@@ -63,26 +63,6 @@ workflows:
...
-  it4:
-    jobs:
-    - gradle/test:
-          <<: *it-defaults
-          test_command: --no-daemon -Dorg.gradle.daemon=false --info it4
-  it5:
-    jobs:
-    - gradle/test:
-          <<: *it-defaults
-          test_command: --no-daemon -Dorg.gradle.daemon=false --info it5
```
Comment (shaikhu) [inline .circleci/config.yml:85]: Once breaking changes with gradle 8 are fixed, we can add back more gradle versions for integration testing.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (guillermo-varela) [reply gradle/wrapper/gradle-wrapper.properties:3]: > If we want to upgrade to 8.8 we should tackle it in anothor PR.

Agreed!

File: gradle.properties (hunk 6 above)
Comment (guillermo-varela) [reply gradle.properties:21]: We can aim to support CycloneDX 1.5 instead in another PR.

Comment (guillermo-varela) [reply gradle.properties]: Also, upgrading to `5.1.2-01` seems quite straightforward:
11982125ad17b7793b921dd605cecb90a71c4104

I'd advocate to update to that latest version.

**Review by guillermo-varela — APPROVED** [review]
+1

File: gradle.properties (hunk 6 above)
Comment (shaikhu) [reply gradle.properties]: > Also, upgrading to `5.1.2-01` seems quite straightforward:
> [1198212](https://github.com/sonatype-nexus-community/scan-gradle-plugin/commit/11982125ad17b7793b921dd605cecb90a71c4104)

Nice!

**Pull request merged**